newrelic/internal/transform/metric.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package transform
+
+import (
+	"errors"
+
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/metric"
+	metricsdk "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+)
+
+// ErrUnimplementedAgg is returned when a transformation of an unimplemented
+// aggregator is attempted.
+var ErrUnimplementedAgg = errors.New("unimplemented aggregator")
+
+// Record transforms an OpenTelemetry Record into a Metric.
+//
+// An ErrUnimplementedAgg error is returned for unimplemented Aggregators.
+func Record(service string, r metricsdk.Record) (telemetry.Metric, error) {
+	desc := r.Descriptor()
+	attrs := attributes(service, desc, r.Labels())
+	switch a := r.Aggregator().(type) {
+	case aggregator.MinMaxSumCount:
```

Comment:
is this a complete enumeration of aggregator types at this point, or is this just a starting point?

Reply:
Is the only ones currently exported. [There are a few other aggregations](https://github.com/open-telemetry/opentelemetry-go/tree/master/sdk/metric/aggregator), but these are the only two used for all (3) current instruments.